### PR TITLE
Adding rubocop to ruby layer

### PR DESCRIPTION
**Pull request by dcluna**
[rubocop](https://github.com/bbatsov/rubocop) is a useful lint tool and should have its place in Spacemacs.

### CODE REVIEW

**Comment by robbyoconnor** [comment]
Awhile ago I would have objected but this is handy...Flycheck can run rubocop for you ***BUT*** this is handy for the autocorrect :D

File: layers/!lang/ruby/README.org [hunk 1]
```diff
@@ -87,3 +90,14 @@ a couple of useful keybindings:
...
 | ~SPC m t b~ | run test file       |
 | ~SPC m t t~ | run test at pointer |
+
+** rubocop
+
+| Key binding   | Description                   |
+|---------------+-------------------------------|
+| ~SPC m r a f~ | check current file            |
+| ~SPC m r a d~ | check current directory       |
+| ~SPC m r a p~ | check current project         |
```
Comment (syl20bnr) [inline layers/!lang/ruby/README.org]: do we need the check commands ? Can flycheck do it automatically for us ?

Comment (robbyoconnor) [inline layers/!lang/ruby/README.org]: I was opposed to #2362 on these grounds, but literally the only saving grace for this is the autocorrect functionality...I agree with @syl20bnr on this one...leave the autocorrect stuff though, that's handy.

Comment (dcluna) [inline layers/!lang/ruby/README.org]: OK, should I remove these? How do I do it? Just push a new commit, send a new PR or something else?

Comment (robbyoconnor) [inline layers/!lang/ruby/README.org]: commit and push to the same branch and then rebase (choose "fixup") -- your PR should be ***ONE*** commit.

Comment (dcluna) [inline layers/!lang/ruby/README.org]: Thanks, should be good to go now.

**Comment by robbyoconnor** [comment]
:+1: <3

**Comment by robbyoconnor** [comment]
A quick 'git pull --rebase` on this branch should bring it up to date with the current `develop` ;)

**Comment by nixmaniack** [comment]
Apart from rebase we will need hooks/keys for ruby-mode since we moved away from enh-ruby as default.

**Comment by robbyoconnor** [comment]
> Apart from rebase we will need hooks/keys for ruby-mode since we moved away from enh-ruby as default.

yes, please do this @dcluna  ;)

**Comment by syl20bnr** [comment]
Thank you :+1: 
I adapted it in commit https://github.com/syl20bnr/spacemacs/commit/5b99a60 the main change is that RuboCop bindings are under `SPC m r r`.
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**